main.tf

```diff
@@ -155,12 +155,19 @@ resource "aws_route_table" "private" {
 }
 
 resource "aws_route" "private" {
-  count                  = var.enable_nat_gateway && local.private_subnets > 0 && local.public_subnets > 0 ? local.private_subnets : 0
+  count                  = !var.enable_endpoint_egress_routing && var.enable_nat_gateway && local.private_subnets > 0 && local.public_subnets > 0 ? local.private_subnets : 0
   route_table_id         = aws_route_table.private[count.index].id
-  destination_cidr_block = var.private_subnet_default_route
+  destination_cidr_block = "0.0.0.0/0"
   nat_gateway_id         = aws_nat_gateway.default[count.index].id
 }
 
+resource "aws_route" "endpoint_egress_routing" {
+  count                  = var.enable_endpoint_egress_routing && local.private_subnets > 0 ? length(var.endpoint_egress_data) : 0
+  route_table_id         = var.endpoint_egress_data[count.index].route_table_id
+  destination_cidr_block = var.endpoint_egress_data[count.index].destination_cidr_block
+  vpc_endpoint_id        = var.endpoint_egress_data[count.index].vpc_endpoint_id
+}
```

Comment:
Wouldn't it be easier to not include this in this module? I would suggest to _only_ make the default route optional. Any other customer specific rules can be added outside of this module in the customer environment.

Reply:
makes sense, I'll refactor to optionalize default route.